diff --git a/framework/bass/system.go b/framework/bass/system.go
index 7ab29d8..26493b0 100644
--- a/framework/bass/system.go
+++ b/framework/bass/system.go
@@ -56,7 +56,7 @@ func Init(offscreen bool) {
 	// BASS_CONFIG_MP3_OLDGAPS
 	C.BASS_SetConfig(C.DWORD(68), C.DWORD(1))
 
-	deviceId := -1 //default audio device
+	deviceId := 1 //default audio device
 	mixerFlags := C.BASS_MIXER_NONSTOP
 
 	if offscreen {
diff --git a/framework/env/paths.go b/framework/env/paths.go
index 07cec35..ffb2175 100644
--- a/framework/env/paths.go
+++ b/framework/env/paths.go
@@ -16,8 +16,9 @@ func Init(pkgName string) {
 	execPath := GetExecDir()
 
 	execPathLower := strings.ToLower(execPath)
+	isRunningInNix := strings.HasPrefix(execPathLower, "/nix/store")
 
-	if strings.HasPrefix(execPathLower, "/usr/bin") || strings.HasPrefix(execPathLower, "/usr/lib") { //if pkgName is a package
+	if strings.HasPrefix(execPathLower, "/usr/bin") || strings.HasPrefix(execPathLower, "/usr/lib") || isRunningInNix { //if pkgName is a package
 		homePath, err := os.UserHomeDir()
 		if err != nil {
 			panic(err)
@@ -42,6 +43,10 @@ func Init(pkgName string) {
 		}
 
 		libDir = "/usr/lib/" + pkgName
+
+		if isRunningInNix {
+			libDir = execPath;
+		}
 	} else {
 		libDir = execPath
 		dataDir = execPath

